eHealthAfrica/ckanext-biskit
# control debug link display

# Description

Introduce a configuration variable to control the display of the debug link within the footer when running the portal in DEBUG mode.

**Rational:**
The BISKIT portal production instance is to be run in `DEBUG` mode. Though not ideal, that's the mode able to serve necessary vendor static resources (css and javascript) for a properly functioning portal. In non-DEBUG mode, static resources are not present and lead to pages with forms not exhibiting appropriate state/behaviour.

This is a [known issue](https://github.com/ckan/ckan/issues/4934) without a solution. Thus, this is a stop-gap solution, pending when a solution is found.

Config setting:
```ini
ckanext.biskit.show_debug_link = (true|false)
```

## Todos

- [ ] Unit tests added (if appropriate)

## How to test
Set the following config (env) vars inside the `.env` file:

```ini
CKAN___DEBUG = true
CKANEXT__BISKIT__SHOW_DEBUG_LINK = false
```

Run the setup, the debug link should be missing in the footer.

Changing the settings as below should restore the debug link back in the footer.

```ini
CKAN___DEBUG = true
CKANEXT__BISKIT__SHOW_DEBUG_LINK = true
```

## Screenshots and/or Gifs
N/A

## Associated JIRA Tasks
N/A

## Known Issues
N/A
